.github/workflows/composer-validate.yml

```diff
@@ -0,0 +1,29 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Composer Validate
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
```

Comment:
Could be only
```suggestion
on: pull_request
```

Reply:
And we need to guarantee that no one will commit on main branch

Reply:
This is made as setting at repository.